### ISSUE TITLE
when a hunk is discarded in GitKraken from a watched file, the watcher process crashes

### ISSUE THREAD

**Issue by catdad** [catdad/watchboy#20]
```
D:\Git\raw-viewer\node_modules\watchboy\index.js:167
      const { evname, evarg } = pending[abspath];
              ^

TypeError: Cannot destructure property `evname` of 'undefined' or 'null'.
    at Timeout.pending.(anonymous function).timer.setTimeout [as _onTimeout] (D:\Git\raw-viewer\node_modules\watchboy\index.js:167:40)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

**Comment by catdad** [comment]
Update: after some code changes, it looks like this error still happens but it a bit different:

```
D:\Git\fair-compare\node_modules\watchboy\index.js:182
      pending[abspath].timer = -1;
                             ^

TypeError: Cannot set property 'timer' of undefined
    at Timeout.pending.(anonymous function).timer.setTimeout [as _onTimeout] (D:\Git\fair-compare\node_modules\watchboy\index.js:182:30)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

**Comment by catdad** [comment]
This should be fixed in `watchboy@0.4.3`
